source/locationHelper.py

```diff
@@ -0,0 +1,398 @@
+#locationHelper.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited, Babbage B.V.
+
+"""Classes and helper functions for working with rectangles and coordinates."""
+
+from collections import namedtuple
+import windowUtils
+import winUser
+from ctypes.wintypes import RECT, SMALL_RECT, POINT
+import textInfos
+import wx
+
+class Point(namedtuple("Point",("x","y"))):
+	"""Represents a point on the screen."""
+
+	def __add__(self,other):
+		"""Returns a new L{Point} with its coordinates representing the additions of the original x and y coordinates."""
+		if not isinstance(other,POINT_CLASSES):
+			return NotImplemented
+		return Point((self.x+other.x),(self.y+other.y))
+
+	def __radd__(self,other):
+		"""Returns a new L{Point} with x = self.x + other.x and y = self.y + other.y."""
+		return self.__add__(other)
+
+	def __sub__(self,other):
+		if not isinstance(other,POINT_CLASSES):
+			return NotImplemented
+		return Point((self.x-other.x),(self.y-other.y))
+
+	def __rsub__(self,other):
+		return self.__sub__(other)

```

Comment:
Could you elaborate? For add and radd, we have the sum function, but I belief there is no such function for subtraction, right?

Reply:
Hmm, I didn't think through how this might get called. Can you answer that, how would this get called? Why is it necessary? Is there some case (as with `__rsum__`) that it might get called with `other` as an integer?

Reply:
This would get called if you do ctypes.wintypes.POINT(3,3) + locationHelper.Point(4,4), in which case it should return locationHelper.Point(7,7)

Reply:
Sorry, I didn't follow that. My questions were about the `__rsub__` function. I found the following on the [python docs](https://docs.python.org/2.0/ref/numeric-types.html)
> These functions are called to implement the binary arithmetic operations (`+`, `-`, `*`, `/`, `%`, `divmod()`, `pow()`, `**`, `«`, `»`, `&`, `^`, `|`) with reversed operands. These functions are only called if the left operand does not support the corresponding operation. For instance, to evaluate the expression `x-y`, where `y` is an instance of a class that has an `__rsub__()` method, `y.__rsub__(x)` is called. Note that ternary `pow()` will not try calling `__rpow__()` (the coercion rules would become too complicated).

So adding a test to show that: 
`ctypes.wintypes.POINT(3,3) - locationHelper.Point(1,1)` is `locationHelper.Point(2,2)`
and also perhaps a test that shows that:
`0 - locationHelper.Point(1,1)` will raise an unsupported exception.

However, this raises a question: what happens in the case of negation:
`-locationHelper.Point(1,1)`
Does this return `locationHelper.Point(-1,-1)?
A test to demonstrate this outcome would be good.